### PR TITLE
feat(RDS): add a new resource: rds backup

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run=TestAccBackup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run=TestAccBackup_basic -timeout 360m -parallel 4
=== RUN   TestAccBackup_basic
=== PAUSE TestAccBackup_basic
=== CONT  TestAccBackup_basic
--- PASS: TestAccBackup_basic (759.54s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       759.587s
```

### CODE REVIEW

File: docs/resources/rds_backup.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  backed up.  
+  (Only Microsoft SQL Server supports partial backups.).
+
+  Changing this parameter will create a new resource.
+The [BackupDatabase](#Backup_BackupDatabase) structure is documented below.
+
+<a name="Backup_BackupDatabase"></a>
+The `BackupDatabase` block supports:
+
+* `name` - (Optional, String) Database to be backed up for Microsoft SQL Server.
```
Comment (ShiChangkuo) [inline docs/resources/rds_backup.md]: it seems that this parameter is mandatory and can not be update

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+		},
+	}
+	return &sc
+}
+
+func resourceBackupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	// createBackup: create a Cloud Connect.
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_backup.go]: **Cloud Connect**?

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+			}
+
+			if utils.StrSliceContains(strings.Split(`COMPLETED`, ","), status.(string)) {
+				return createBackupWaitingRespBody, "COMPLETED", nil
+			}
+
+			return createBackupWaitingRespBody, "PENDING", nil
+		},
+		Timeout:      t,
+		Delay:        5 * time.Second,
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_backup.go]: is 5 seconds too short? suggest extending the delay time.

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+			createBackupWaitingRespBody, err := utils.FlattenResponse(createBackupWaitingResp)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+			status, err := jmespath.Search(`backups[0].status`, createBackupWaitingRespBody)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error parse %s from response body", `backups[0].status`)
+			}
+
+			if utils.StrSliceContains(strings.Split(`FAILED`, ","), status.(string)) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_backup.go:260]: it's recommended to get the string array in magic template.

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/rds/resource_huaweicloud_rds_backup.go]: ok, change to 30

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/rds/resource_huaweicloud_rds_backup.go]: fixed

File: docs/resources/rds_backup.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/rds_backup.md]: ok，fixed

File: huaweicloud/services/rds/resource_huaweicloud_rds_backup.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/rds/resource_huaweicloud_rds_backup.go:260]: ok, change  template later

**Pull request merged**